src/Sfa.Das.ApprenticeshipInfoService.Core/Helpers/PaginationHelper.cs

```diff
@@ -0,0 +1,29 @@
+﻿using SFA.DAS.Apprenticeships.Api.Types.Pagination;
+
+namespace Sfa.Das.ApprenticeshipInfoService.Core.Helpers
+{
+    public class PaginationHelper : IPaginationHelper
+    {
+        public PaginationDetails GeneratePaginationDetails(int page, int pageSize, int totalCount)
+        {
+            var paginationDetails = new PaginationDetails { NumberOfRecordsToSkip = 0, Page = 1, TotalCount = totalCount, NumberPerPage = pageSize };
+
+            if (totalCount <= 0)
+            {
+                return paginationDetails;
+            }
+
+            var pageUsed = page <= 0 ? 1 : page;
+            var skip = (pageUsed - 1) * pageSize;
+            while (skip >= totalCount)
+            {
+                skip = skip - pageSize;
+            }
+
+            paginationDetails.LastPage = (totalCount + pageSize - 1) / pageSize;
+            paginationDetails.NumberOfRecordsToSkip = skip;
+            paginationDetails.Page = (skip / pageSize) + 1;
+            return paginationDetails;
+        }
```

Comment:
The only thing that is used here is the pages to skip value. 
Would it be possible that this method only returns that value?

Reply:
The story requires the maximum page number to be available in the front end - there is also front end work to do in das-search that will consume this api. In addition, the pagesize to use is stored as a value in the api rather than the front end, so there needed to be a mechanism available to inform the front end various details, like what the last page is, what page you are on, how many pages you records you've skipped, what the page size is, and so on.

I rolled these up into the paginationDetails class